src/System.Private.DataContractSerialization/src/System/Runtime/Serialization/DataContract.cs

```diff
@@ -699,8 +710,6 @@ private static DataContract CreateDataContract(int id, RuntimeTypeHandle typeHan
                                     dataContract = new CollectionDataContract(type);
                                 else if (type.GetTypeInfo().IsEnum)
                                     dataContract = new EnumDataContract(type);
-                                else if (Globals.TypeOfException.IsAssignableFrom(type))
```

Comment:
Good point. I opened https://github.com/dotnet/corefx/issues/13099 for revisiting this later.